managed/services/backup/removal_service.go

```diff
@@ -78,91 +81,106 @@ func (s *RemovalService) DeleteArtifact(ctx context.Context, artifactID string,
 			ArtifactID: artifactID,
 		})
 		if err != nil {
+			s.setFailedToDeleteBackupStatus(artifactID)
 			return err
 		}
 
 		for _, ri := range restoreItems {
 			if err := models.RemoveRestoreHistoryItem(tx.Querier, ri.ID); err != nil {
+				s.setFailedToDeleteBackupStatus(artifactID)
 				return err
 			}
 		}
 
-		return models.DeleteArtifact(tx.Querier, artifactID)
+		err = models.DeleteArtifact(tx.Querier, artifactID)
+		if err != nil {
+			s.setFailedToDeleteBackupStatus(artifactID)
```

Comment:
Let's add log here, otherwise we losing error message.

Reply:
Changed logic.